frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/LegacyFeatureToggleSwitch.tsx

```diff
@@ -0,0 +1,257 @@
+import { UPDATE_FEATURE_ENVIRONMENT } from 'component/providers/AccessProvider/permissions';
+import { useOptimisticUpdate } from './hooks/useOptimisticUpdate';
+import { flexRow } from 'themes/themeStyles';
+import { ENVIRONMENT_STRATEGY_ERROR } from 'constants/apiErrors';
+import { formatUnknownError } from 'utils/formatUnknownError';
+import useFeatureApi from 'hooks/api/actions/useFeatureApi/useFeatureApi';
+import { useFeature } from 'hooks/api/getters/useFeature/useFeature';
+import useToast from 'hooks/useToast';
+import { useChangeRequestsEnabled } from 'hooks/useChangeRequestsEnabled';
+import { useChangeRequestToggle } from 'hooks/useChangeRequestToggle';
+import { EnableEnvironmentDialog } from './EnableEnvironmentDialog/EnableEnvironmentDialog';
+import { UpdateEnabledMessage } from 'component/changeRequest/ChangeRequestConfirmDialog/ChangeRequestMessages/UpdateEnabledMessage';
+import { ChangeRequestDialogue } from 'component/changeRequest/ChangeRequestConfirmDialog/ChangeRequestConfirmDialog';
+import {
+    FeatureStrategyProdGuard,
+    useFeatureStrategyProdGuard,
+} from '../../../../feature/FeatureStrategy/FeatureStrategyProdGuard/FeatureStrategyProdGuard';
+
+const StyledBoxContainer = styled(Box)<{ 'data-testid': string }>(() => ({
+    mx: 'auto',
+    ...flexRow,
+}));
+
+interface IFeatureToggleSwitchProps {
+    featureId: string;
+    environmentName: string;
+    projectId: string;
+    value: boolean;
+    onError?: () => void;
+    onToggle?: (
+        projectId: string,
+        feature: string,
+        env: string,
+        state: boolean,
+    ) => void;
+}
+
+/**
+ * @deprecated remove when flag `featureSwitchRefactor` is removed
+ */
+export const FeatureToggleSwitch: VFC<IFeatureToggleSwitchProps> = ({
+    projectId,
+    featureId,
+    environmentName,
+    value,
+    onToggle,
+    onError,
+}) => {
+    const { loading, toggleFeatureEnvironmentOn, toggleFeatureEnvironmentOff } =
+        useFeatureApi();
+    const { setToastData, setToastApiError } = useToast();
+    const { isChangeRequestConfigured } = useChangeRequestsEnabled(projectId);
+    const {
+        onChangeRequestToggle,
+        onChangeRequestToggleClose,
+        onChangeRequestToggleConfirm,
+        changeRequestDialogDetails,
+    } = useChangeRequestToggle(projectId);
+    const [isChecked, setIsChecked, rollbackIsChecked] =
+        useOptimisticUpdate<boolean>(value);
+
+    const [showEnabledDialog, setShowEnabledDialog] = useState(false);
+    const { feature } = useFeature(projectId, featureId);
+    const enableProdGuard = useFeatureStrategyProdGuard(
+        feature,
+        environmentName,
+    );
+    const [showProdGuard, setShowProdGuard] = useState(false);
+
+    const disabledStrategiesCount =
+        feature?.environments
+            .find((env) => env.name === environmentName)
+            ?.strategies.filter((strategy) => strategy.disabled).length ?? 0;
+
+    const handleToggleEnvironmentOn = async (
```

Comment:
Should follow the ADR of `onAction` for handlers

Reply:
`LegacyFeatureToggleSwitch.tsx`, not new code